src/linear_layer.cpp

```diff
@@ -463,6 +463,17 @@ void linear_bwd_fc_delta_b_mp(std::vector<float> &var_b,
     }
 }
 
+void linear_update_hidden_states_worker(float &mu_a_prior, float &var_a_prior,
+                                        std::vector<float> &delta_mu,
+                                        std::vector<float> &delta_var,
+                                        float &mu_a_post, float &var_a_post)
+/*
+ */
+{
+    mu_a_post = mu_a_prior + delta_mu[0] * var_a_prior;
+    var_a_post = (1.0f + delta_var[0] * var_a_prior) * var_a_prior;
+}
+
```

Comment:
I would separate the Linear from the smoother option e.g., creating another SmootherLinear layer similar to Linear.

Reply:
Ok. I will do it.

Reply:
solved by using SLinear layer